Shared/Samples/Filter features in scene/FilterFeaturesInSceneView.swift

```diff
@@ -0,0 +1,243 @@
+    }
+    
+    /// The different states for filtering features in a scene.
+    enum FilterState: Equatable {
+        case load, filter, reset
+        
+        /// A human-readable label for the filter state.
+        var label: String {
+            switch self {
+            case .load: return "Load"
+            case .filter: return "Filter"
+            case .reset: return "Reset"
+            }
+        }
+        
+        /// The next filter state to apply to a scene.
+        func next() -> Self {
+            switch self {
+            case .load:
+                return .filter
+            case .filter:
+                return .reset
+            case .reset:
+                return .load
+            }
+        }
+    }
+}
+
+private extension FilterFeaturesInSceneView {
+    /// The model used to store the geo model and other expensive objects
+    /// used in this view.
+    class Model: ObservableObject {
+        /// The scene for this sample.
+        let scene: ArcGIS.Scene
+        
+        /// The open street map layer for the sample.
+        private let osmBuildings: ArcGISSceneLayer
+        
+        /// The San Francisco buildings scene layer for the sample.
+        private let buildingsSceneLayer = ArcGISSceneLayer(
+            url: URL(string: "https://tiles.arcgis.com/tiles/z2tnIkrLQ2BRzr6P/arcgis/rest/services/SanFrancisco_Bldgs/SceneServer")!
+        )
+        
+        /// The graphics overlay for the scene view.
+        let graphicsOverlay = GraphicsOverlay()
+        
+        /// A polygon that shows the extent of the detailed buildings scene layer.
+        private let polygon: ArcGIS.Polygon = makeFilteringPolygon()
+        
+        /// A red extent boundary graphic that represents the full extent of the detailed buildings scene layer.
+        private var sanFranciscoExtentGraphic: Graphic {
+            let simpleFillSymbol = SimpleFillSymbol(
+                style: .solid,
+                color: .clear,
+                outline: SimpleLineSymbol(
+                    style: .solid,
+                    color: .red,
+                    width: 5
+                )
+            )
+            
+            return Graphic(
+                geometry: polygon,
+                symbol: simpleFillSymbol
+            )
+        }
+        
+        init() {
+            // Create basemap.
+            let vectorTiledLayer = ArcGISVectorTiledLayer(
+                item: PortalItem(
+                    portal: .arcGISOnline(connection: .anonymous),
+                    id: .osmTopographic
+                )
+            )
+            osmBuildings = ArcGISSceneLayer(
+                item: PortalItem(
+                    portal: .arcGISOnline(connection: .anonymous),
+                    id: .osmBuildings
+                )
+            )
+            let basemap = Basemap()
+            basemap.addBaseLayers([vectorTiledLayer, osmBuildings])
```

Comment:
```suggestion
            let basemap = Basemap(baseLayers: [vectorTiledLayer, osmBuildings])
```
Up to you, can also move the `osmBuildings` out as a [default property value](https://docs.swift.org/swift-book/documentation/the-swift-programming-language/initialization/#Default-Property-Values).